Fix fatal error on product page

wjrosa opened the pull request:
Fixes https://github.com/Automattic/woocommerce-subscriptions-core/pull/718#issuecomment-2479441073

## Description

<!--
Write a brief summary about this PR. 
- Why is this change needed? 
- What does this change do? 
- Were there other solutions you considered? 
- Why did you choose to pursue this solution? 
- Describe any trade-offs you might have had to make.
-->

This PR fixes a fatal error thrown on the product page reported by @annemirasol [here](https://github.com/Automattic/woocommerce-subscriptions-core/pull/718#issuecomment-2479441073). Since the list of products can be made of IDs, I will be keeping the original implementation here (before https://github.com/Automattic/woocommerce-subscriptions-core/pull/718)

<!--
Questions for PR author:
- How can this code break?
- What are we doing to make sure this code doesn't break?
-->

<!--
Images or gifs: Include before and after screenshots or gifs/videos as appropriate.
-->

## How to test this PR

<!--
Testing instructions: How should this be tested and how can a reviewer test the end-user functionality? Are there known issues that you plan to address in a future PR? Are there any side effects that readers should be aware of?
Use the testing instructions from the linked issue as a starting point.
-->

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

Code review. You can also reproduce the issue following the steps highlighted by Anne [here](https://github.com/Automattic/woocommerce-subscriptions-core/pull/718#issuecomment-2479441073).

## Product impact
<!-- What products will this PR ship in? -->

- [ ] Added changelog entry (or does not apply)
- [ ] Will this PR affect WooCommerce Subscriptions? yes/no/tbc, add issue ref
- [ ] Will this PR affect WooCommerce Payments? yes/no/tbc, add issue ref
- [ ] <!-- 🚨 Deprecations 🚨 --> Added deprecated functions, hooks or classes to the [spreadsheet](https://docs.google.com/spreadsheets/d/1xw9xszcPMnWsp4C8OKZMsLzZob7tOmWT7qMqmEIq314/edit#gid=0)

cc @annemirasol

wjrosa committed:
Fix fatal error on product page

wjrosa committed:
Reverting to original implementation

mattallan reviewed (approved):
Thanks @wjrosa, changes LGTM :+1:

Another option would be to add:

```
if ! ( $child_product ) {
    continue;
}
```

after the `wc_get_product()` to confirm we got a valid object back. I'm fine with reverting this change though to avoid the fatal.

wjrosa committed:
Reverting meta retrieval method + adding extra check for child product instance

wjrosa commented:
> Thanks @wjrosa, changes LGTM 👍
> 
> Another option would be to add:
> 
> ```
> if ! ( $child_product ) {
>     continue;
> }
> ```
> 
> after the `wc_get_product()` to confirm we got a valid object back. I'm fine with reverting this change though to avoid the fatal.

Nice, Matt! I reverted the method removal and added a similar check:
```
if ( ! $child_product instanceof WC_Product ) {
				continue;
			}
```
